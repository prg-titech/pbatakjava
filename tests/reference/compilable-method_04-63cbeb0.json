{
    "basename": "compilable-method_04-63cbeb0",
    "cmd": "{infile}",
    "infile": "tests/method_04/build.sh",
    "infile_hash": "0b9fdedb398d89d484f396ad318abd0ca5bdcb802296bcf5293d7a9b",
    "stdout": "compilable-method_04-63cbeb0.stdout",
    "stdout_hash": "20a1da2b78fca4eafea76d634fa9c5ba74c13d53b48cb60d216eda4a",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}