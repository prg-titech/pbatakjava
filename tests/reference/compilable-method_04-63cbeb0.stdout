
public interface Point {
  public int get__x();

  public int set__x(int toAssign);

  public int get__y();

  public int set__y(int toAssign);

  public Point inverse(Point_Factory V);

  public Point mirror(Point_Factory V);
}

public interface Point_Factory {
  public Point make(int x, int y);
}

public class Point_v2 implements Point {
  public int x;

  public int y;

  public Point_v2(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public Point inverse(Point_Factory V) {
    return V.make(y, x);
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }

  public Point mirror(Point_Factory V) {
    return null;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v2(x, y);
  }
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public Point mirror(Point_Factory V) {
    return V.make(y, x);
  }

  public int get__x() {
    return this.x;
  }

  public int set__x(int toAssign) {
    return this.x = toAssign;
  }

  public int get__y() {
    return this.y;
  }

  public int set__y(int toAssign) {
    return this.y = toAssign;
  }

  public Point inverse(Point_Factory V) {
    return null;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }
}
